specification/schemaregistry/data-plane/Microsoft.EventHub/preview/2018-01-01-preview/schemaregistry.json

```diff
@@ -1,132 +1,461 @@
-      "name": "schemaregistryClient"
-    }
-  },
-  "host": "management.azure.com",
-  "schemes": ["https"],
-  "consumes": ["application/json"],
-  "produces": ["application/json"],
-  "security": [
-    {
-      "azure_auth": ["user_impersonation"]
-    }
-  ],
-  "securityDefinitions": {
-    "azure_auth": {
-      "type": "oauth2",
-      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
-      "flow": "implicit",
-      "description": "Azure Active Directory OAuth2 Flow",
-      "scopes": {
-        "user_impersonation": "impersonate your user account"
-      }
-    }
+    "title": "Azure Schema Registry",
+    "version": "0.1.9"
   },
   "paths": {
-    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.EventHub/operations": {
+    "/$schemagroups/getSchemaById": {
       "get": {
-        "tags": ["Tag1"],
-        "operationId": "OperationGroup_Get",
-        "x-ms-examples": {
-          "BatchAccountDelete": { "$ref": "./examples/OperationGroupGet.json" }
-        },
-        "description": "This is a sample get operation, please see guidelines in azure-rest-api-specs repository for more info",
+        "summary": "Get schema by GUID",
+        "description": "Get schema by GUID.  Azure Schema Registry guarantees that GUID within a namespace.",
+        "operationId": "getSchemaById",
+        "produces": [
+          "application/json"
+        ],
         "parameters": [
           {
-            "$ref": "#/parameters/SubscriptionIdParameter"
+            "in": "header",
+            "type": "string",
+            "name": "X-Serialization",
+            "description": "schema type",
+            "required": true
+          },
+          {
+            "in": "header",
+            "type": "string",
+            "name": "X-Schema-Id",
+            "description": "unique schema identifier",
+            "required": true
           },
           {
-            "$ref": "#/parameters/ResourceGroupNameParameter"
+            "in": "header",
+            "type": "string",
+            "name": "X-Schema-Id-Location",
+            "description": "location of schema resource",
+            "required": true
+          },
+          {
+            "in": "header",
+            "type": "string",
+            "name": "X-Schema-Version",
+            "description": "version of returned schema",
+            "required": true
           }
         ],
         "responses": {
           "200": {
-            "description": "Describe the result of a successful operation.",
+            "description": "OK",
             "schema": {
-              "$ref": "#/definitions/Result"
+              "$ref": "#/definitions/SchemaBytePayload"
             }
           },
-          "default": {
-            "description": "Error response describing why the operation failed.",
+          "404": {
+            "description": "Schema with matching GUID not found"
```

Comment:
This won't raise an exception, since the default semantic of Swagger is that declared status code are not errors. You could either remove it (and let the SDK deal with it, if you have server message that should be clear), or use the x-ms-client-error to explain to autorest that you want an exception.
Given you don't have a schema, I would remove it (unless you don't want a 404 to raise an exception, but that's the usual pattern)

Reply:
Let's try removing it.  In the Java SDK, we had to manually check the response status code and throw if it matched 404, which wasn't ideal.

Reply:
I'm going to just remove all 4xx status codes.